lib/Auth/Process/EnsureVoMember.php

```diff
@@ -0,0 +1,170 @@
+    {
+        parent::__construct($config, $reserved);
+        $config = Configuration::loadFromArray($config);
+
+        if (is_null($config)) {
+            throw new Exception(
+                'perun:EnsureVoMember: Property  \'' . self::ENSURE_VO_MEMBER . '\' is missing or invalid!'
+            );
+        }
+
+        $this->triggerAttr = $config->getString(self::TRIGGER_ATTR, "");
+
+        if (empty($this->triggerAttr)) {
+            throw new Exception(
+                'perun:EnsureVoMember: Missing configuration option \'' . self::TRIGGER_ATTR . '\''
+            );
+        }
+
+        $this->voDefsAttr = $config->getString(self::VO_DEFS_ATTR, "");
+
+        if (empty($this->voDefsAttr)) {
+            throw new Exception(
+                'perun:EnsureVoMember: Missing configuration option \'' . self::VO_DEFS_ATTR . '\''
+            );
+        }
+
+        $this->loginUrlAttr = $config->getString(self::LOGIN_URL, null);
+        $this->registrarUrl = $config->getString(self::REGISTRAR_URL, null);
+
+        $interface = $config->getString(self::INTERFACE_PROPNAME, self::RPC);
+        $this->adapter = Adapter::getInstance($interface);
+    }
+
+    public function process(&$request)
+    {
+        if (isset($request['SPMetadata']['entityid'])) {
+            $spEntityId = $request['SPMetadata']['entityid'];
+        } else {
+            throw new Exception(
+                'perun:EnsureVoMember: Cannot find entityID of remote SP. ' .
+                'hint: Do you have this filter in IdP context?'
+            );
+        }
+
+        if (isset($request['perun']['user'])) {
+            $user = $request['perun']['user'];
+        } else {
+            throw new Exception(
+                'perun:EnsureVoMember: ' .
+                'missing mandatory field \'perun.user\' in request.' .
+                'Hint: Did you configured PerunIdentity filter before this filter?'
+            );
+        }
+
+        $facility = $this->adapter->getFacilityByEntityId($spEntityId);
+
+        if ($facility === null) {
+            Logger::debug('perun:EnsureVoMember: skip execution - no facility provided');
+            return;
+        }
+
+        $attrValues = $this->adapter->getFacilityAttributesValues(
+            $facility,
+            [$this->voDefsAttr, $this->triggerAttr, $this->loginUrlAttr]
+        );
+
+        $triggerAttrValue = $attrValues[$this->triggerAttr];
+        if ($triggerAttrValue === null || $triggerAttrValue === false) {
+            Logger::debug(
+                'perun:EnsureVoMember: skip execution - attribute ' . self::TRIGGER_ATTR . ' is null or false'
+            );
+            return;
+        }
+
+        if (empty($attrValues[$this->voDefsAttr])) {
+            Logger::debug(
+                'perun:EnsureVoMember: skip execution - attribute ' . self::VO_DEFS_ATTR . ' has null or no value'
+            );
+            return;
+        }
+        $voShortName = $attrValues[$this->voDefsAttr];
```

Comment:
to stay consistent with the code above, I would first do the assignment to the variable and then use it in the If statement :)